src/electron/platform/android/appium-service-configurator.ts

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import ADB from 'appium-adb';
+import {
+    AndroidServiceConfigurator,
+    DeviceInfo,
+    PackageInfo,
+    PermissionInfo,
+} from 'electron/platform/android/android-service-configurator';
+
+type AdbDevice = {
+    udid: string;
+};
+
+const servicePackageName: string = 'com.microsoft.accessibilityinsightsforandroidservice';
+
+export class AppiumServiceConfigurator implements AndroidServiceConfigurator {
+    private readonly adb: ADB;
+
+    constructor(adb: ADB) {
+        this.adb = adb;
+    }
+
+    public getConnectedDevices = async (): Promise<Array<DeviceInfo>> => {
+        // First add devices that are flagged as emulator, then
+        // add devices that aren't flagged as emulators
+        const detectedDevices = {};
+        await this.addDevices(await this.adb.getConnectedEmulators(), true, detectedDevices);
+        await this.addDevices(await this.adb.getConnectedDevices(), false, detectedDevices);
+
+        const output: Array<DeviceInfo> = [];
+        Object.values(detectedDevices).forEach(device => output.push(device as DeviceInfo));
+        return output;
+    };
+
+    private async addDevices(
+        devices: Array<AdbDevice>,
+        isEmulator: boolean,
+        detectedDevices: any,
+    ): Promise<void> {
+        for (let loop = 0; loop < devices.length; loop++) {
```

Comment:
A foreach loop would match the intent a little more closely (`for (const device of devices)` is the syntax)

Reply:
I agree, and originally wrote this using foreach. Unfortunately, foreach doesn't understand the await syntax and causes the function to exit without waiting. I'll add a comment to that effect

Reply:
Were you perhaps trying to uses a loop like `devices.forEach((device) => { /* stuff */ });`? It's true that that wouldn't work as-is, but you should be able to use the statement form of foreach loop with async/await (`for (const device of devices) { /* stuff */ }`).

Reply:
Thanks--updated